app/views/contexts/preview/preview_question.py

```diff
@@ -0,0 +1,119 @@
+from typing import Any, Optional, Union
+
+from app.data_models import QuestionnaireStore
+from app.questionnaire import Location, QuestionnaireSchema, QuestionSchemaType
+from app.questionnaire.placeholder_renderer import PlaceholderRenderer
+from app.questionnaire.variants import transform_variants
+
+
+class PreviewQuestion:
+    def __init__(
+        self,
+        schema: QuestionnaireSchema,
+        questionnaire_store: QuestionnaireStore,
+        section_id: str,
+        block_id: str,
+        language: str,
+    ):
+        self.schema = schema
+        self.questionnaire_store = questionnaire_store
+        self.current_location = Location(section_id=section_id, block_id=block_id)
+        self.block_id = block_id
+
+        self.placeholder_renderer = PlaceholderRenderer(
+            language=language,
+            answer_store=self.questionnaire_store.answer_store,
+            list_store=self.questionnaire_store.list_store,
+            metadata=self.questionnaire_store.metadata,
+            response_metadata=self.questionnaire_store.response_metadata,
+            schema=self.schema,
+            location=self.current_location,
+            preview_mode=True,
+        )
+
+        self.question = self.rendered_block().get("question")
```

Comment:
We should render the section up front and pass the rendered json instead of getting this far then rendering each time with a new instance of a placeholder renderer?. That way you only need one placeholder renderer and you only render the section once?.

Reply:
Ok whole section rendered upfront and then just stripping down to single block when being passed across the classes. This reduced the amount of code significantly.